src/rust/ide/lib/view/src/graph_editor/component/documentation_view.rs

```diff
@@ -0,0 +1,159 @@
+        DocumentationViewModel{dom,logger,size}.init()
+    }
+
+    fn init(self) -> Self {
+        self.reload_style();
+        self
+    }
+
+    fn set_size(&self, size:Vector2) {
+        self.size.set(size);
+        self.reload_style();
+    }
+
+    const PLACEHOLDER_STR: &'static str = "<h3>Enso Documentation Viewer</h3>\
+                                           <p>No documentation available</p>";
+
+    fn receive_data(&self, data:&Data) -> Result<(),DataError> {
+        let data_inner = match data {
+            Data::Json {content} => content,
+            _                    => todo!(),
+        };
+
+        let data_str = serde_json::to_string_pretty(&**data_inner);
+        let data_str = data_str.unwrap_or_else(|e| format!("<Cannot render data: {}>", e));
+        // Fixes a Doc Parser Bug - to be removed when rewritten to rust
+        let data_str = data_str.replace("\\n", "\n");
+        let data_str = data_str.replace("\"", "");
+
+        let parser = parser::DocParser::new_or_panic();
+        let output = parser.generate_html_doc_pure(data_str);
+        let output = output.unwrap_or_else(|_| String::from(DocumentationViewModel::PLACEHOLDER_STR));
+        // Fixes a Doc Parser related idea, where stylesheet was a separate file
+        let output = output.replace(r#"<link rel="stylesheet" href="style.css" />"#, "");
+
+        let data_str = format!(r#"<div class="docVis">{}{}</div>"#, get_doc_style(), output);
+        self.dom.dom().set_inner_html(&data_str);
+        Ok(())
+    }
+
+    /// Generates welcome screen HTML.
+    pub fn welcome_screen(&self) {
+        let placeholder = DocumentationViewModel::PLACEHOLDER_STR;
+        let data_str    = format!(r#"<div class="docVis">{}{}</div>"#, get_doc_style(), placeholder);
+        self.dom.dom().set_inner_html(&data_str)
+    }
+
+    fn reload_style(&self) {
+        self.dom.set_size(self.size.get());
+    }
+}
+
+// =========================
+// === DocumentationView ===
+// =========================
+
+/// Visualization that renders the given documentation as a HTML page
+#[derive(Clone,CloneRef,Debug,Shrinkwrap)]
+#[allow(missing_docs)]
+pub struct DocumentationView {
+    #[shrinkwrap(main_field)]
+    pub model : DocumentationViewModel,
+    pub frp   : visualization::instance::Frp,
+    network   : frp::Network,
+}
+
+impl DocumentationView {
+    /// Definition of this visualization.
+    pub fn definition() -> Definition {
+        let path = Path::builtin("Documentation View Visualization (native)");
+        Definition::new(
+            Signature::new_for_any_type(path,Format::Json),
+            |scene| { Ok(Self::new(scene).into()) }
+        )
+    }
+
+    /// Constructor.
+    pub fn new(scene:&Scene) -> Self {
+        let network = default();
+        let frp   = visualization::instance::Frp::new(&network);
+        let model = DocumentationViewModel::new(scene);
+        model.welcome_screen();
```

Comment:
functions should have semantic naming. If I see `model.welcome_screen()` I completely do not know what this function does? Do it prints welcome screen on paper, closes it, displays it, toggles it, inits it? Whati t does?